### PR TITLE
[https://nvbugs/6693991][fix] Restore verl rollout after sampler_type removal

### PR DESCRIPTION
**Pull request by zhaoyangwang-nvidia**
## Description

`77cc145384` (#18232) removed `TRTLLMSampler` and the `sampler_type` argument. That was a breaking change for callers outside this repository, and one of them is pinned by our own CI: `tests/integration/defs/verl/verl_config.yml` pins verl at `d324b01`, whose rollout server passes `sampler_type` unconditionally (`trtllm_async_server.py:248`). Every verl rollout test then failed constructing the engine, which is the post-merge failure in nvbugs/6693991.

The important detail is that `sampler_type` is not an independent knob on the verl side. A single switch, `TLLM_USE_TORCHSAMPLER` (default `"0"`, and never set in this repo), selects both the sampler and the configuration that matches it:

| `TLLM_USE_TORCHSAMPLER` | `sampler_type` | `end_id` / `detokenize` | `logprobs` |
|---|---|---|---|
| `1` | `TorchSampler` | resolved eos / `True` | `0` |
| unset (`0`) | `TRTLLMSampler` | `-1` / `False` | `1` |

So simply letting the argument through is not sufficient, and is not safe: it would run TorchSampler against the C++ sampler's conventions. `logprobs=1` returns the top-1 log probability, whereas TorchSampler reports the sampled token's at `logprobs=0` — for an RL rollout that is a wrong training signal rather than a crash, and the warning would only reach a Ray actor log.

This PR therefore does two things:

- **`verl_config.yml`**: set `TLLM_USE_TORCHSAMPLER=1`, moving verl's whole profile onto TorchSampler so the sampler, stop tokens and logprobs conventions agree. This is what actually makes the configuration correct.
- **`TorchLlmArgs`**: accept `sampler_type` only for the values that still resolve to TorchSampler (`auto`, `TorchSampler`) and drop them with a warning; reject `TRTLLMSampler` with an error naming the settings to review. Removed arguments are declared once in `_TORCH_LLMARGS_REMOVED_ARG_CHECKS`, which the two pre-pydantic kwarg gates in `llm.py` and the model validator all consult, so adding another removed argument later cannot leave the gates and the validator out of sync.

No field is reintroduced, so the API stability references are unaffected. Note that verl `main` still passes the argument (487 commits ahead of the pin), so bumping `repo_tag` alone would not have fixed this; dropping it upstream is worth a follow-up.

## Test Coverage

- `verl/test_verl_cases.py` — all six cases waived under nvbugs/6693991 are unwaived here: `test_unimodal_generate`, `test_unimodal_batch_generate`, `test_multimodal_generate_with_image`, `test_multimodal_different_image_sizes`, `test_multimodal_text_only_fallback`, `test_wake_sleep_cycle`. They are the entire contents of verl's `test_trtllm_rollout_utils.py` and share class-scoped fixtures that call `TRTLLMReplica.init_standalone()` -> `launch_servers()` -> `AsyncLLM(**llm_kwargs)` with no mocking and no branch that can skip it, so they fail and recover together.
- These live in the `post_merge` / `backend: verl` stage (4xB200), so they need `/bot run --extra-stage "..."` to run pre-merge.
- The remaining waived verl cases are unrelated: nvbugs/6683838 and nvbugs/6272653 were filed before this removal (2026-08-28 and 2026-06-05, vs 77cc145384 on 2026-08-30), so they keep their waivers.

## PR Checklist

Please review the following before submitting your PR:
- PR description clearly explains what and why. If using CodeRabbit's summary, please make sure it makes sense.
- PR Follows [TRT-LLM CODING GUIDELINES](https://github.com/NVIDIA/TensorRT-LLM/blob/main/CODING_GUIDELINES.md) to the best of your knowledge.
- Test cases are provided for new code paths (see [test instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/tests#1-how-does-the-ci-work))
- If PR introduces API changes, an appropriate PR label is added - either `api-compatible` or `api-breaking`. For `api-breaking`, include `BREAKING` in the PR title.
- Any new dependencies have been scanned for license and vulnerabilities
- [CODEOWNERS](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/CODEOWNERS) updated if ownership changes
- Documentation updated as needed
- Update [tava architecture diagram](https://github.com/NVIDIA/TensorRT-LLM/blob/main/.github/tava_architecture_diagram.md) if there is a significant design change in PR.
- The reviewers assigned automatically/manually are appropriate for the PR.

- [x] Please check this after reviewing the above items as appropriate for this PR.

## Dev Engineer Review

- Restores Verl rollout compatibility with `TorchSampler`.
- Enables `TLLM_USE_TORCHSAMPLER=1` in the Verl CI configuration.
- Accepts `sampler_type="auto"` and `"TorchSampler"` with warnings.
- Rejects `"TRTLLMSampler"` and unsupported values with actionable errors.
- Centralizes removed-argument validation.
- Updates sampling documentation.
- Removes six obsolete Verl waivers.

## QA Engineer Review

- Updates `tests/integration/defs/verl/verl_config.yml` to enable `TorchSampler`.
- Removes six entries from `tests/integration/test_lists/waives.txt`.
- The affected Verl tests run in the post-merge `backend: verl` stage.
- Verdict: sufficient.

### CODE REVIEW

**Review by BowenFu — APPROVED** [review]

**Comment by zhaoyangwang-nvidia** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-Verl-Post-Merge-1" --disable-fail-fast

**Review by LarryXFly — APPROVED** [review]

**Comment by zhaoyangwang-nvidia** [comment]
/bot run --extra-stage "DGX_B200-4_GPUs-Verl-Post-Merge-1" --disable-fail-fast

**Review by chzblych — APPROVED** [review]

**Comment by zhaoyangwang-nvidia** [comment]
Hi @allisonlim-nv since junyi is on vacation could help to review this PR, thanks~

**Review by arysef — APPROVED** [review]

**Review by lori-ren — APPROVED** [review]

**Comment by zhaoyangwang-nvidia** [comment]
/bot run --disable-fail-fast

Both failures in pipeline #58056 look unrelated to this PR (which only touches `sampler_type` compatibility in `llm_args.py`/`llm.py` plus verl test config/waivers):

- **`DGX_B200-PyTorch-8`** - `test_e2e.py::test_ptp_quickstart_advanced_ngram[Llama-3.1-8B-Instruct]` hit a pytest Timeout (Slurm job 1844978 FAILED, node gpu-46). The identical signature (`Infra / Timeout`, same Slurm cluster) also occurred in builds #58013 (PR #14606) and #58032 (PR #17466) - ~3 timeouts out of ~28 runs in this window across three unrelated PRs. The reported "Test terminated unexpectedly" is the stage-timeout placeholder, not a real assertion.
- **`H100_PCIe-PyTorch-Perf-1`** - `test_perf[llama_v3.1_8b_instruct-bench-pytorch-float16-input_output_len:128,128-reqs:8192]`: `trtllm-bench` was SIGKILLed (9) after ~3524s on node ipp2-2362, with no Python exception or assertion in the captured output - consistent with an OOM/stall kill rather than a code failure. This stage passed in 24/24 other builds in the window, so re-running with `--disable-fail-fast` to confirm.

`Test-x86_64-Multi-GPU` was blocked as a cascading effect of the x86 single-GPU failure, not a separate root cause.
